app/main.py

```diff
@@ -2,5 +2,17 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_results = {}
+
+    def wrapper(*args, **kwargs) -> int:
```

Comment:
You should indicate that `wrapper` returns `Any` because we do not know in advance which function the cache decorator will be applied to.